### DISCUSSION TITLE
Support for OSM (Open Service Mesh)

### DISCUSSION THREAD

**Discussion by phillipgibson** [kiali/kiali#4166] [Roadmap]
The OSM team is looking at developing a UI, but want to consider utilizing Kiali and extending Kiali to additional service mesh projects beyond Istio. We think this can be a great community collaboration.

**Comment by jmazzitelli** [comment]
(fwiw: for past discussions on this kind of thing, see https://github.com/kiali/kiali/issues/2823 and https://github.com/kiali/kiali/issues/1679 and https://github.com/kiali/kiali/issues/3778)

**Reply by phillipgibson** [reply]
Just looked at those issues posted. I think we're willing to invest the engineering time for the effort and collaborate that way. Maybe we can look to possibly extend the SMI Spec for this so all other meshes that utilize SMI can benefit as well.

**Reply by phillipgibson** [reply]
How do we do an RFC for this work to integrate?

**Reply by israel-hdez** [reply]
We usually use a shared Google document and comment on it. There is no template, it's completely free form.

**Reply by lucasponce** [reply]
Hi @phillipgibson, thanks for your request, the team is moving to use the github discussions for design topics, so, feel free to create a first high level version like a discussion.
In that way conversations and comments are easily trackable.
Then if some area require more deep details, a shared document, or any other open format is also welcome.

Discussions are the previous step to create Epics or Issues, and allow the team to interact better.

**Comment by jshaughn** [comment]
Hi @phillipgibson, let's dig a little deeper here...

As you likely know, Kiali works with multiple sources of data: Istio telemetry and config,  Kubernetes config and logs, tracing data, etc.  If Kiali were to support multiple service meshes it would likely require creation of proper abstraction layers, with intermediate models to adapt features to various sources, or hide features not yet supported by an underlying source.  The investment for this approach would likely require a dedicated small/mid-sized engineering team for decent period of time (probably min 6-12months), meanwhile other teams would continue to move forward the project.

We'd like to know a bit more what you are thinking for Kiali with OSM support?  Are you thinking about an offering that has mostly a parity with Kiali-Istio?  Or possibly something that offers a specific subset of Kiali-Istio features?

Perhaps we could schedule a community call to talk about these questions and answers, and to arrive at a more formal proposal?

Thanks

**Reply by phillipgibson** [reply]
Totally makes sense. We were combing through the code and looking at the informers etc. on how the project was fetching data. Our early thoughts is that maybe the abstraction lives in the SMI spec, therefore all service meshes that utilize the spec can benefit from this as well as supporting their APIs for the project.

**Comment by lucasponce** [comment]
Next July 30th is planned the Kiali Community meeting.
I think this discussion could be added as part of the agenda for that day giving visibility to the community of users if they would like to participate as well.

**Reply by xeviknal** [reply]
Agree :) 

@PhilipGough I've just opened the agenda for the next community meeting: https://github.com/kiali/kiali/discussions/4186
The meeting starts with demos of what will be new in the next release and we approach topics proposed by the community afterwards. 

Do you mind adding the topic(s) you'd like to present in the discussion above?

If you've got more technical low-level doubts, I guess you can either keep adding comments in this discussion or opening [a new discussion](https://github.com/kiali/kiali/discussions/new).

**Reply by philipgough** [reply]
@xeviknal I'm going to assume you meant to tag @phillipgibson - if not the answer is going to be a no from me :)

**Reply by xeviknal** [reply]
@PhilipGough yes indeed. Sorry for the noise. Anyway, you are more than welcome to step in whenever you feel, haha.
We should open a PR into github to better sort the autocomplete dropdowns, hehe. It was a clear shot here.

**Reply by abonas** [reply]
@phillipgibson just bumping this in case this got lost in a lot of replies - would you like to join to the next community meeting and discuss this?

**Reply by phillipgibson** [reply]
@abonas sorry I was out on vacation for the last two weeks. I'll get on the next call to discuss

**Reply by xeviknal** [reply]
@phillipgibson thank you for letting us know.
Here you have the community calendar as well: http://bit.ly/kiali-calendar
